sdk/src/main/scala/com/horizen/account/state/AccountStateView.scala

```diff
@@ -340,7 +355,25 @@ class AccountStateView(
 
   // TODO: get baseFee for the block header
   // TODO: currently a non-zero baseFee makes all the python tests fail, because they do not consider spending fees
-  override def getBaseFee: BigInteger = BigInteger.valueOf(0)
+  override def getBaseFeePerGas: BigInteger = BigInteger.valueOf(333000)
+
+  def getTxFeesPerGas(tx: EthereumTransaction) : (BigInteger, BigInteger) = {
+    if (tx.isEIP1559) {
+      val baseFeePerGas = getBaseFeePerGas
+      val maxFeePerGas = tx.getMaxFeePerGas
+      val maxPriorityFeePerGas = tx.getMaxPriorityFeePerGas
+      // if the Base Fee plus the Max Priority Fee exceeds the Max Fee, the Max Priority Fee will be reduced
+      // in order to maintain the upper bound of the Max Fee.
+      val forgerTipPerGas = if (baseFeePerGas.add(maxPriorityFeePerGas).compareTo(maxFeePerGas) > 0) {
+        maxFeePerGas.subtract(baseFeePerGas)
+      } else {
+        maxPriorityFeePerGas
+      }
+      (baseFeePerGas, forgerTipPerGas)
+    } else {
+      (BigInteger.ZERO, tx.getGasPrice)
```

Comment:
The type of transaction that a user submits has an impact on the way we distribute fees? Is this how we want it to be?
Even in legacy transactions the gasPrice has to be greater or equal to the base fee, why don't we just divide the gasPrice into base fee and tip and distribute fees accordingly?
`(baseFeePerGas, tx.getGasPrice.subtract(baseFeePerGas))`

Reply:
I took inspiration from what I saw in geth/etherscan, where there is no burnt quota for legacy txes. I think both alternatives are viable, and I personally like best the approach you are proposing. Lets discuss it further.

Reply:
Interesting. I had a look too and on Etherscan it really looks like legacy TXs do not burn any of the fees, but that is not true. The geth implementation does not differentiate between TX types [here](https://github.com/ethereum/go-ethereum/blob/20356e57b119b4e70ce47665a71964434e15200d/core/state_transition.go#L335).
```
effectiveTip := st.gasPrice
if london {
    effectiveTip = cmath.BigMin(st.gasTipCap, new(big.Int).Sub(st.gasFeeCap, st.evm.Context.BaseFee))
}
st.state.AddBalance(st.evm.Context.Coinbase, new(big.Int).Mul(new(big.Int).SetUint64(st.gasUsed()), effectiveTip))
```
At this point, for any type of TX there are usable values for all the fields (gasPrice, feeCap and tip). So this works on any kind of TX as expected: burn the base fee, everything else goes to coinbase. Same thing is applicable to us because we do the same when creating a message from a TX: if it is a legacy TX set both feeCap and tip to gasPrice.

Here are some very helpful illustrations I just found:
- [EIP-1559 TX](https://uploads-ssl.webflow.com/5fe512bf3e57baa22a4812f0/611fec11baefcfd52512b216_0*R_f7IHC0jAQBJBEC.png)
- [Legaxy TX](https://uploads-ssl.webflow.com/5fe512bf3e57baa22a4812f0/611fec114bafd083ee330450_0*nu522iCGRurPooVF.png)
- [Source article](https://www.liquity.org/blog/how-eip-1559-changes-the-transaction-fees-of-ethereum)

Reply:
Ok, I added fee splitting pool/forger also for legacy txes